src/components/v2/V2Project/V2PayButton/V2PayForm/V2PayForm.tsx

```diff
@@ -0,0 +1,158 @@
+import { SmileOutlined } from '@ant-design/icons'
+import * as constants from '@ethersproject/constants'
+import { t, Trans } from '@lingui/macro'
+import { Checkbox, Form, Input, Space, Switch, Tooltip } from 'antd'
+import { FormInstance, useWatch } from 'antd/lib/form/Form'
+import { EthAddressInput } from 'components/inputs/EthAddressInput'
+import { FormImageUploader } from 'components/inputs/FormImageUploader'
+import { AttachStickerModal } from 'components/modals/AttachStickerModal'
+import { ThemeContext } from 'contexts/themeContext'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+import { isAddress } from 'ethers/lib/utils'
+import { useContext, useState } from 'react'
+
+import { ProjectPreferences } from 'constants/v2/projectPreferences'
+import { StickerSelection } from './StickerSelection'
+
+export interface V2PayFormType {
+  memo?: string
+  beneficiary?: string
+  stickerUrls?: string[]
+  uploadedImage?: string
+  preferClaimed?: boolean
+}
+
+export const V2PayForm = ({ form }: { form: FormInstance<V2PayFormType> }) => {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+
+  const { tokenAddress } = useContext(V2ProjectContext)
+
+  const [customBeneficiaryEnabled, setCustomBeneficiaryEnabled] =
+    useState<boolean>(false)
+
+  const [attachStickerModalVisible, setAttachStickerModalVisible] =
+    useState<boolean>(false)
+
+  const hasIssuedTokens = tokenAddress !== constants.AddressZero
+
+  const stickerUrls = useWatch('stickerUrls', form)
+
+  const canAddMoreStickers =
+    (stickerUrls ?? []).length < ProjectPreferences.STICKER_MAX
+
+  return (
+    <>
+      <Form form={form} layout="vertical">
+        <Form.Item
+          label={t`Memo (optional)`}
+          className={'antd-no-number-handler'}
+          extra={t`Add an on-chain memo to this payment.`}
+        >
+          <Form.Item noStyle name="memo">
```

Comment:
**question**: i think we still have [`MemoFormItem.tsx`](https://github.com/jbx-protocol/juice-interface/blob/290fe0226f09f13c3f59fa0d1ab039641c1450cf/src/components/inputs/Pay/MemoFormItem.tsx). Is it worth trying to move any new code into that component, and using an instance of it here?

Reply:
Also TIL `noStyle` prop

Reply:
We could yeah - only problem with that class is it doesn't conform to what antd expect (value/onChange) so it is bit of extra work. I think MemoFormItem is only used in v1 too. I am happy to do it though if you think it is valuable to update v1 code

Reply:
nvm - it actually does conform to it. I'll look at moving

Reply:
Actually I think it is more complicated than initially seemed. I think if we want to reuse it, lets revisit then. MemoFormItem is only used in v1 pay atm.

Reply:
We were using MemoFormItem in V2 https://github.com/jbx-protocol/juice-interface/pull/1349/files#r915355021

It would be nice to re-use in V1 and V2 if possible. Otherwise we're duplicating code. We can leave this for a follow-up PR though - really want to get this PR in as-is.

Reply:
yeah lets do it - don't mind following up in separate PR
https://github.com/jbx-protocol/juice-interface/issues/1354
❤️